### PR TITLE
Hide unnecessary profile sections for bots

### LINKED ISSUE #6147: Profiles like BanchoBot still need to become similar to the web version
**Describe the bug:** (You're probably aware of this, but I would like it to report to see if anyone already wants to implement it)
If Lazer "pages" have to be similar to web, there are a few problems with profiles like [**BanchoBot**](https://osu.ppy.sh/users/3).

**Screenshots or videos showing encountered issue:** 
* The _Saint Helena_ flag is missing on Lazer:
web
Lazer

* And on the web-side, the page only has the _me!_ tab (still missing on Lazer, but I guess it's a future thing), but on Lazer everything is visible (like _total plays_ on the left and all other tabs like _Recents_, _Ranks_, _Kudosu!_ etc.)
web
Lazer

**osu!lazer version:** 2019.913.0

### PR DESCRIPTION
**Pull request by EVAST9919**
Fixes https://github.com/ppy/osu/issues/6147

### CODE REVIEW

**Comment by peppy** [comment]
How are you testing this?

**Comment by iiSaLMaN** [comment]
Also when adding test, please make it useful for CI, by having a way to ensure the sections are not added to the container when the loaded user is a bot.

File: osu.Game.Tests/Visual/Online/TestSceneUserProfileOverlay.cs [hunk 1]
```diff
@@ -106,6 +106,14 @@ protected override void LoadComplete()
                 Country = new Country { FullName = @"Japan", FlagName = @"JP" },
                 CoverUrl = @"https://osu.ppy.sh/images/headers/profile-covers/c6.jpg"
             }, api.IsLoggedIn));
+            AddStep("Show bancho", () => profile.ShowUser(new User
```
Comment (bdach) [inline osu.Game.Tests/Visual/Online/TestSceneUserProfileOverlay.cs]: CI's not complaining, but for consistency with the other steps and to separate them out visually I'd add a blank line before this.

**Review by bdach — COMMENTED** [review]
Might have been nice to add an `Assert` step to check that the irrelevant profile sections aren't loaded after `ShowUser` with `IsBot = true`, but that would require to make them at least `protected`, so I'm unsure if it's worth the hassle of breaking encapsulation to test it.

**Comment by iiSaLMaN** [comment]
> But that would require to make them at least `protected`, so I'm unsure if it's worth the hassle of breaking encapsulation to test it.

It's worth it and would make the tests useful for CI, we set the accessibility of the field to `protected` and expose it with a subclass of it `TestMyComponent` if it cannot be accessed from anywhere else, but for this one, the sections can be exposed like this `overlay.OfType<ProfileSectionsContainer>().First().OfType<ProfileSections>()`.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**